sdk/helper/certutil/helpers.go

```diff
@@ -1756,6 +1780,25 @@ func parseCertificateToFields(certificate x509.Certificate) (templateData map[st
 	return templateData, nil
 }
 
+func getBasicConstraintsFromExtension(exts []pkix.Extension) (found bool, isCA bool, maxPathLength int, err error) {
+	for _, ext := range exts {
+		if ext.Id.Equal(ExtensionBasicConstraintsOID) {
+			type basicConstraints struct {
```

Comment:
Could you use `ParseBasicConstraintExtension` here?

Reply:
Switch is made, that's a much cleaner piece of code, thank you!